src/HasMedia/HasMediaTrait.php

```diff
@@ -282,31 +298,14 @@ public function addMediaConversion($name)
     }
 
     /**
-     * Delete the model. The extra logic isn't handled in a model event since the boot function is unreliable
-     * for currently unknown reasons.
-     *
-     * @return bool
-     */
-    public function delete()
-    {
-        if (!parent::delete()) {
-            return false;
-        }
-
-        $this->media()->get()->map(function ($media) {
-            $media->delete();
-        });
-
-        return true;
-    }
-
-    /**
      * Delete the model, but preserve all the associated media.
      *
      * @return bool
      */
     public function deletePreservingMedia()
     {
+        $this->deletePreservingMedia = true;
+
         return parent::delete();
```

Comment:
No it could not. Because `deleted` event will be fired in this case and you will not be able to prevent media deletion. I've done so because unit tests were red.